extensions/wikia/UserLogin/FacebookSignupController.class.php

```diff
@@ -53,6 +53,12 @@ public function index() {
 				$user->setCookies();
 				$this->loggedIn = true;
 				$this->userName = $user->getName();
+
+				// Retrieve user email from Facebook if missing
+				$email = $user->getEmail();
+				if ( empty( $email ) ) {
+					$this->saveEmailAsynchronously( $user->getId() );
```

Comment:
I am aware of this scenario, and that is why I log all attempts, so that we have records of them all. If we see that this is a big issue, we can adjust it later on.

Making an additional round-trip to FB for every user missing an email on every signup just for checking email and doing so online is not very enticing for an alternative. We should not block user interaction especially when they login and are ready to contribute and engage within a Wikia community.